openassessment/workflow/models.py

```diff
@@ -287,6 +287,17 @@ def update_from_assessments(self, assessment_requirements):
 
         step_for_name = {step.name: step for step in steps}
 
+        # If the status is done or cancelled, check if score has changed.
+        if self.status == self.STATUS.done:
+            if force_update_score:
+                new_score = self.get_score(assessment_requirements, step_for_name)
+                self.set_score(new_score)
+                self.save()
+                logger.info((
+                    u"Workflow for submission UUID {uuid} has updated score."
+                ).format(uuid=self.submission_uuid))
+            return
```

Comment:
That does seem like a more general solution but it will need a couple of tweaks towards the bottom of the function too (because the score is only set if the current status is `waiting`). Let me look into this.